apps/src/gamelab/spritelab/Spritelab.js

```diff
@@ -0,0 +1,218 @@
+    commands.edgesDisplace.apply(this, [spriteId]);
+  };
+
+  this.commands.isTouchingEdges = function(spriteId) {
+    return commands.isTouchingEdges.apply(this, [spriteId]);
+  };
+
+  this.commands.jumpTo = function(spriteId, location) {
+    commands.jumpTo(spriteId, location);
+  };
+
+  this.commands.mirrorSprite = function(spriteId, direction) {
+    commands.mirrorSprite(spriteId, direction);
+  };
+
+  this.commands.moveInDirection = function(spriteId, distance, direction) {
+    commands.moveInDirection(spriteId, distance, direction);
+  };
+
+  this.commands.moveForward = function(spriteId, distance) {
+    commands.moveForward(spriteId, distance);
+  };
+
+  this.commands.moveToward = function(spriteId, distance, target) {
+    commands.moveToward(spriteId, distance, target);
+  };
+
+  this.commands.removeTint = function(spriteId) {
+    commands.setProp(spriteId, 'tint', null);
+  };
+
+  this.commands.setProp = function(spriteId, prop, val) {
+    commands.setProp.apply(this, [spriteId, prop, val]);
+  };
+
+  this.commands.setTint = function(spriteId, color) {
+    commands.setProp(spriteId, 'tint', color);
+  };
+
+  this.commands.turn = function(spriteId, n, direction) {
+    commands.turn(spriteId, n, direction);
+  };
+
+  // Behavior commands
+  this.commands.addBehaviorSimple = function(spriteId, behavior) {
+    commands.addBehavior(spriteId, behavior);
+  };
+
+  this.commands.Behavior = function(callback) {
+    return commands.Behavior(callback);
+  };
+
+  this.commands.draggableFunc = function(spriteId) {
+    return commands.draggableFunc(this);
+  };
+
+  this.commands.removeAllBehaviors = function(spriteId) {
+    commands.removeAllBehaviors(spriteId);
+  };
+
+  this.commands.removeBehaviorSimple = function(spriteId, behavior) {
+    commands.removeBehavior(spriteId, behavior);
+  };
+
+  // Event commands
+  this.commands.checkTouching = function(
+    condition,
+    sprite1,
+    sprite2,
+    callback
+  ) {
+    commands.checkTouching(condition, sprite1, sprite2, callback);
+  };
+
+  this.commands.keyPressed = function(condition, key, callback) {
+    commands.keyPressed(condition, key, callback);
+  };
+
+  this.commands.spriteClicked = function(condition, spriteId, callback) {
+    commands.spriteClicked(condition, spriteId, callback);
+  };
+
+  // Location commands
+  this.commands.locationAt = function(x, y) {
+    return commands.locationAt(x, y);
+  };
+
+  this.commands.locationMouse = function() {
+    return commands.locationMouse.apply(this);
+  };
+
+  this.commands.locationOf = function(spriteId) {
+    return commands.locationOf(spriteId);
+  };
+
+  this.commands.randomLocation = function() {
+    return commands.randomLocation();
+  };
+
+  // Sprite commands
+  this.commands.countByAnimation = function(animation) {
+    return commands.countByAnimation(animation);
+  };
+
+  this.commands.createNewSprite = function(name, animation, location) {
```

Comment:
Looks like name isn't used here.

Reply:
Yeah this is another quirk of generated code :/
This block generates:
```
var name;
name = createNewSprite(name, animation, location);
```
I don't know of a way to make it *not* pass name into the function as well.
It's probably possible, but I don't want to block this change on that. I'll log a ticket in our backlog though to clean this up for all assignment blocks.

Reply:
No prob. Add a comment so someone else doesn't delete it :)

Reply:
👍